### ISSUE TITLE
Wrong error diagnostic for pyspark

### ISSUE THREAD

**Issue by tgkei** [microsoft/pyright#5487]
**Describe the bug**
When try to use [pyspark col function](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.functions.col.html), pyright diagnose it as error while it's not

**To Reproduce**
install pyspark and use col function. In my case, I imported `functions` through `import pyspark.sql.functions as F`, and used it like `F.col("col_name")`.

**Expected behavior**
Do not diagnose it as error. 

**Code or Screenshots**
![image](https://github.com/microsoft/pyright/assets/37530453/bdefdca6-3cb9-4b52-b441-c9b04db2d023)
```python
import pyspark.sql.functions as F

F.col("col1") # diagnose it as error
```

pyspark version is 3.0.3.
That code is run on pyright-language server (through nvim)
I don't know specific version of pyright, but I believe it is the newest, since I installed it last week.

**Comment by erictraut** [comment]
I'm not able to repro the problem with the latest version of pyspark (3.4.1). I suspect you simply need to update pyspark. In any case, this is unlikely to be a bug in pyright. Pyright simply operates on the type information provided by the library.
